test/test_torch.py

```diff
@@ -17257,6 +17264,38 @@ def test_nextafter(self, device, dtype):
         expected = np.nextafter(a.cpu().numpy(), b.cpu().numpy())
         self.assertEqual(actual, expected, atol=0, rtol=0)
 
+    @dtypesIfCUDA(torch.float16, torch.float32, torch.float64)
+    @dtypesIfCPU(torch.bfloat16, torch.float32, torch.float64)
+    @dtypes(torch.float32, torch.float64)
+    @unittest.skipIf(not TEST_SCIPY, "SciPy not found")
+    def test_i0(self, device, dtype):
```

Comment:
This is a solid test but I'd like to hear your thinking about i0's numerics before discussing testing further.